src/modules/mavlink/mavlink_messages.cpp

```diff
@@ -4299,9 +4145,9 @@ static const StreamListItem streams_list[] = {
 	StreamListItem(&MavlinkStreamCommandLong::new_instance, &MavlinkStreamCommandLong::get_name_static, &MavlinkStreamCommandLong::get_id_static),
 	StreamListItem(&MavlinkStreamSysStatus::new_instance, &MavlinkStreamSysStatus::get_name_static, &MavlinkStreamSysStatus::get_id_static),
 	StreamListItem(&MavlinkStreamHighresIMU::new_instance, &MavlinkStreamHighresIMU::get_name_static, &MavlinkStreamHighresIMU::get_id_static),
-	StreamListItem(&MavlinkStreamScaledIMU::new_instance, &MavlinkStreamScaledIMU::get_name_static, &MavlinkStreamScaledIMU::get_id_static),
-	StreamListItem(&MavlinkStreamScaledIMU2::new_instance, &MavlinkStreamScaledIMU2::get_name_static, &MavlinkStreamScaledIMU2::get_id_static),
-	StreamListItem(&MavlinkStreamScaledIMU3::new_instance, &MavlinkStreamScaledIMU3::get_name_static, &MavlinkStreamScaledIMU3::get_id_static),
+	StreamListItem(&MavlinkStreamScaledIMU<0>::new_instance, &MavlinkStreamScaledIMU<0>::get_name_static, &MavlinkStreamScaledIMU<0>::get_id_static),
+	StreamListItem(&MavlinkStreamScaledIMU<2>::new_instance, &MavlinkStreamScaledIMU<2>::get_name_static, &MavlinkStreamScaledIMU<2>::get_id_static),
+	StreamListItem(&MavlinkStreamScaledIMU<3>::new_instance, &MavlinkStreamScaledIMU<3>::get_name_static, &MavlinkStreamScaledIMU<3>::get_id_static),
```

Comment:
I believe that this misses the IMU with ID 1, since you pass the N to the subscriber.

Reply:
Thanks @mhkabir,

You are correct, it is unfortunately true that SCALED_IMU2 polls accel1 and gyro 1. :) I appreciate your review.

(NOTE: At present, using PX4/Firmware master branch, even without adding the additional SCALED_IMU2 and SCALED_IMU3 messages to the 'onboard' list and using QGC and the mavlink inspector, even the current SCALED_IMU instance is not visible.  I can see that the onboard stream is being sent as I can see SYSTEM_TIME message coming through which is not part of the normal stream.  There appear to be a number of additional messages missing in the mavlink inspector view as well right now and it has made verifying this PR a bit tricky.  I am investigating that breakage currently as well.)

Thanks again for reviewing this PR.

-Mark

Reply:
@mhkabir, I'm able to confirm the first SCALED_IMU message is being passed correctly on a pixracer now from this PR.  (SCALED_IMU2 and SCALED_IMU3 don't report through on pixracer because the sensors aren't present.)  I might have a hardware issue with the other board I am testing on that was misdirecting test results.